src/main/java/org/jabref/gui/openoffice/DetectOpenOfficeInstallation.java

```diff
@@ -51,33 +46,34 @@ public DetectOpenOfficeInstallation(JDialog parent, OpenOfficePreferences prefer
         if (this.checkAutoDetectedPaths(preferences)) {
             future.complete(true);
         } else {
-            init();
+            DefaultTaskExecutor.runInJavaFXThread(() -> init());
             BackgroundTask.wrap(() -> future.complete(autoDetectPaths()))
-                          .onSuccess(x -> SwingUtilities.invokeLater(progressDialog::dispose))
+                          .onSuccess(x -> progressDialog.close())
                           .executeWith(Globals.TASK_EXECUTOR);
         }
         return future;
     }
 
-    public void init() {
-        progressDialog = showProgressDialog(parent, Localization.lang("Autodetecting paths..."),
-                Localization.lang("Please wait..."));
+    private void init() {
+        DialogPane dialogPane = new DialogPane();
```

Comment:
Why not reuse the methods in `DialogService`?

Reply:
If you look a few lines later, you see that the DialogPane is passed to the showCustomDialog method ofr the dialogService class

Reply:
is used for showCustomDialog